zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -183,7 +184,7 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             &mut reversion_info,
         );
 
-        let caller_nonce_hash_bytes = cb.query_word32();
+        let caller_nonce_hash_bytes = cb.query_word32(); // What's this variable for?
```

Comment:
This is the new address created with the create opcode. `caller_nonce_hash_bytes ` is a 32 bytes of hash output.

- In line 195. we require the first 20 bytes from it to equal to the newly created callee address.
- In line 215, it is the hash output of the create input.

Reply:
is it equal to keccak(rlp(sender, nonce)) (or in the case of internal call contract creation hash of the memory)?

Reply:
Yes, that's correct.